src/argilla/client/sdk/v1/datasets/models.py

```diff
@@ -93,6 +93,35 @@ class FeedbackRecordsModel(BaseModel):
     total: int
 
 
+# TODO: `query_score` naming can be improved to simply `score`. (frontend should be aligned)
+# TODO: Maybe `query_score` should not be optional.
+class FeedbackRecordSearchModel(BaseModel):
+    record: FeedbackItemModel
+    query_score: Optional[float]
+
+
+class FeedbackRecordsSearchModel(BaseModel):
+    items: List[FeedbackRecordSearchModel]
+    total: int
+
+
+class FeedbackRecordsSearchVectorQuery(BaseModel):
+    name: str
+    record_id: Optional[UUID] = None
+    value: Optional[List[float]] = None
+
+    @root_validator
+    def check_required(cls, values: dict) -> dict:
+        """Check that either 'record_id' or 'value' is provided"""
+        record_id = values.get("record_id")
+        value = values.get("value")
+
+        if bool(record_id) == bool(value):
+            raise ValueError("Either 'record_id' or 'value' must be provided")
+
+        return values
```

Comment:
With the changes mentioned above I think this validation is not required here

Reply:
Let's leave this validation here, since it is aligned with the same schema of the server side. If this validation changes in the future, we can change this code without having to change anything anywhere else